release/src/main/groovy/MoblieGamingJavaUtils.groovy

```diff
@@ -0,0 +1,176 @@
+}
+
+  public static String getUserScoreOutputName(String runner){
+    return "${USERSCORE_OUTPUT_PREFIX}${RUNNERS[runner]}.txt"
+  }
+
+  public static String getHourlyTeamScoreOutputName(String runner){
+    return "${HOURLYTEAMSCORE_OUTPUT_PREFIX}${RUNNERS[runner]}.txt"
+  }
+
+
+  private static String generateUserScoreCommand(String runner, TestScripts t, String commonArgs){
+    StringBuilder cmd = new StringBuilder()
+    StringBuilder exeArgs = new StringBuilder(commonArgs)
+
+    exeArgs.append("--input=gs://${t.gcsBucket()}/5000_gaming_data.csv ")
+    if(runner == "DataflowRunner"){
+      exeArgs.append("--project=${t.gcpProject()} ")
+        .append("--output=gs://${t.gcsBucket()}/${getUserScoreOutputName(runner)} ")
+    } else{
+      exeArgs.append("--output=${getUserScoreOutputName(runner)} ")
+    }
+
+    cmd.append("mvn compile exec:java -q ")
+      .append("-Dexec.mainClass=org.apache.beam.examples.complete.game.UserScore ")
+      .append("-Dexec.args=\"${exeArgs}\" ")
+      .append("-P${RUNNERS[runner]}")
+    return cmd
+  }
+
+
+  private static String generateHourlyTeamScoreCommand(String runner, TestScripts t, String commonArgs){
+    StringBuilder cmd = new StringBuilder()
+    StringBuilder exeArgs = new StringBuilder(commonArgs)
+
+    exeArgs.append("--input=gs://${t.gcsBucket()}/5000_gaming_data.csv ")
+    if(runner == "DataflowRunner"){
+      exeArgs.append("--project=${t.gcpProject()} ")
+        .append("--output=gs://${t.gcsBucket()}/${getHourlyTeamScoreOutputName(runner)} ")
+
+    } else{
+      exeArgs.append("--output=${getHourlyTeamScoreOutputName(runner)} ")
+    }
+
+    cmd.append("mvn compile exec:java -q ")
+      .append("-Dexec.mainClass=org.apache.beam.examples.complete.game.HourlyTeamScore ")
+      .append("-Dexec.args=\"${exeArgs}\" ")
+      .append("-P${RUNNERS[runner]}")
+    return cmd
+  }
+
+
+  private static String generateLeaderBoardCommand(String runner, TestScripts t, String commonArgs){
+    StringBuilder cmd = new StringBuilder()
+    StringBuilder exeArgs = new StringBuilder(commonArgs)
+
+    exeArgs.append("--project=${t.gcpProject()} ")
+      .append("--dataset=${t.bqDataset()} ")
+      .append("--topic=projects/${t.gcpProject()}/topics/${t.pubsubTopic()} ")
+      .append("--output=gs://${t.gcsBucket()}/java-leaderboard-result.txt ")
+      .append("--leaderBoardTableName=leaderboard_${runner} ")
+      .append("--teamWindowDuration=5")
+
+    cmd.append("mvn compile exec:java -q ")
+      .append("-Dexec.mainClass=org.apache.beam.examples.complete.game.LeaderBoard ")
+      .append("-Dexec.args=\"${exeArgs}\" ")
+      .append("-P${RUNNERS[runner]}")
+    return cmd
+  }
+
+
+  private static String generateGameStatsCommand(String runner, TestScripts t, String commonArgs){
+    StringBuilder cmd = new StringBuilder()
+    StringBuilder exeArgs = new StringBuilder(commonArgs)
+
+    exeArgs.append("--project=${t.gcpProject()} ")
```

Comment:
is there any way to take a varargs (or an iterable of args, or probably most preferably a map or multimap) and build this string from that configuration? Making it a map literal seems like it might read more easily.

You could combine that with a method to generate the `exec` command for these examples, and significantly reduce the amount of string-builder specs you have in these methods.